src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -602,13 +602,24 @@ private void CompileSynthesizedMethods(TypeCompilationState compilationState)
                 // any diagnostics emitted into it back into the main diagnostic bag.
                 var diagnosticsThisMethod = DiagnosticBag.GetInstance();
 
-                AsyncStateMachine stateMachineType;
-
                 // Synthesized methods have no ordinal stored in custom debug information (only user-defined methods have ordinals).
                 // In case of async lambdas, which synthesize a state machine type during the following rewrite, the containing method has already been uniquely named, 
                 // so there is no need to produce a unique method ordinal for the corresponding state machine type, whose name includes the (unique) containing method name.
                 const int methodOrdinal = -1;
-                BoundStatement bodyWithoutAsync = AsyncRewriter.Rewrite(methodWithBody.Body, method, methodOrdinal, variableSlotAllocatorOpt, compilationState, diagnosticsThisMethod, out stateMachineType);
+
+                // Previously this was just an async lowering, but local functions can also be iterators, so we need to lower those
```

Comment:
Comments about "Previously" will be confusing to the reader. Best to document the current state of affairs.